### ISSUE TITLE
Model Factory - creating an object and persisting it doesn't always work 

### ISSUE THREAD

**Issue by kfirba** [laravel/lumen-framework#200]
Hello.

I have a really simple code for persisting a user that I use for my tests:

```php
protected function createPredictableRegularUser($overrides = [])
    {
        $overrides = array_merge([
            'type' => 'regular',
            'email' => 'john@example.com'
        ], $overrides);

        factory(User::class)->create($overrides);
        
        // returning an Eloquent object since I need to work with the user's ID as-well
        return User::where($overrides)->firstOrFail();
    }
```

The problem with this is that from time to time I get an error:

```php
Illuminate\Database\Eloquent\ModelNotFoundException: No query results for model [App\User].
```

This error is generated from:

```php
return User::where($overrides)->firstOrFail();
```

Most of the times, it doesn't appear. But from time to time it appears and breaks my tests. Why is that happening? It is as the persisting is happening asynchronously and it may not done persisting it before it's trying to retrieve the user.

**Comment by vluzrmos** [comment]
you could get the return of create method:

```php
protected function createPredictableRegularUser($overrides = [])
    {
        $overrides = array_merge([
            'type' => 'regular',
            'email' => 'john@example.com'
        ], $overrides);

        return factory(User::class)->create($overrides);
   }
```

In that case will be a instance of User, otherwise, if you specify more then 1 ->times(1) it will be a Collection of User.

**Comment by kfirba** [comment]
@vluzrmos have you read my comment? I need the user ID available to me after creating the user. If you inspect the create method you will see that it calls the make() method which will only populate a user object without persisting it and then it will persist the user and return the user that it has populated and not the user from the DB with the ID. 

TL;DR

The user's id isn't available if I return the results from the create() method.

**Comment by vluzrmos** [comment]
If I am right, the `create` method call the `make` method to populate the model and call `save` for each result, the `save` method should override the ID from database.

https://github.com/illuminate/database/blob/master/Eloquent/FactoryBuilder.php#L81

**Comment by kfirba** [comment]
@vluzrmos weird.. Doesn't give me the ID after creating the user... Maybe that's what causing the user to not be found later? The fact that is wasn't saved so there is actually no ID? I'm kinda confused :/

**Comment by vluzrmos** [comment]
weird... it should be something wrong on your attributes.

**Comment by kfirba** [comment]
@vluzrmos I guess.. Unfortunately I find Lumen quite unstable for building a big API. I love the speed it gives you, but I hate the fact that I have to spend hours upon hours in order to find out what's causing a bug that most of the times has something to do with Lumen and not my code. You can see my other issues opened here to see what I'm talking about.

However this particular issue is driving me crazy. My tests can fail just because that for some reason the user isn't being persisted...

**Comment by vluzrmos** [comment]
If your table on database has not an auto-increment ID, you have to set $incrementing=false on model attributes and you have to set manually the ID.

Check out if that is your case.

**Comment by kfirba** [comment]
@vluzrmoa that's not my case :/ I do have an auto-incrementing ID field in my table.

I think that I will just quit using Lumen and just use Laravel while trying to squeeze every bit of performance out of it. If you will take a look at the issues page you can see that I've opened another 3 issues with supposedly bugs (unless I've done something wrong). Hopefully I won't be having this issue with Laravel. They both use the same code base for that but I guess I'm being optimistic :)

**Comment by vluzrmos** [comment]
I know about the log of erros on lumen, its sucks, but on the release >=5.1.3 its changing...

**Comment by kfirba** [comment]
@vluzrmos how do you know that :)? How long would it take until we will see a version which includes all the fixes? I guess that when it will come around I will test it and may migrate my code from Laravel to lumen :)

**Comment by vluzrmos** [comment]
Just listening for commits, on release 5.1.3 some WSOD(White Screens Of Death) are handled.
